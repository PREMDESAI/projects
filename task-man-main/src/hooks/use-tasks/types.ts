export type TaskStatus = 'todo' | 'in_progress' | 'in_review' | 'done';

export interface Task {
  id: string;
  title: string;
  description?: string;
  status: TaskStatus;
}

export interface TaskContextValues {
  isLoading: boolean;
  // Data Layer
  tasksData: Task[];
  // View Layer
  tasksView: Task[];
  statuses: TaskStatus[];

  addTask: (task: Omit<Task, 'id'>) => Promise<void>;
  updateTask: (id: Task['id'], updateTask: Omit<Task, 'id'>) => Promise<void>;
  deleteTask: (id: Task['id']) => Promise<void>;
  filterTasks: (parameters: { text?: string; taskStatus?: TaskStatus }) => void;
}
