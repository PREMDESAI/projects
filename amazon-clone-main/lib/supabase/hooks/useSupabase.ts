import { useState } from "react"
import { supabase } from "../products";

export const useSupabase = () => { 
    const [products, setProducts] = useState<any>([]);
    const [filterData, setFilterData] = useState<any>([]);
    const [singleProduct, setSingleProduct] = useState<any>([]);
    const [mensProduct, setMensProduct] = useState<any>([]);
    const [womensProduct, setWomensProduct] = useState<any>([]);

    const getDataFromSupabase = async () => {
        let {data, error} = await supabase.from('product').select("*");
        if(data){
            setProducts(data); 
        }
        if(error){
            console.log(error);
        }
    }
    const getFilteredData = async (query:string) => {
        let {data, error} = await supabase.from('product').select("*").or(`title.ilike.%${query}%, description.ilike.%${query}%, category.ilike.%${query}%`); // cloth
        if(data){
            setFilterData(data); 
        }
        if(error){
            console.log(error);
        }
    }

    const getSingleProduct = async (id:number)=>{
        let {data, error} = await supabase.from('product').select('*').eq('id', id);
        if(data){
            setSingleProduct(data);
        }
        if(error){
            console.log(error);
        }
    }
    const getMensClothing = async () => {
        let {data, error} = await supabase.from('product').select('*').ilike('category', `men's clothing`);
        if(data){
            setMensProduct(data);
        }
        if(error){
            console.log(error);
            
        }
    }
    const getWomensClothing = async () => {
        let {data, error} = await supabase.from('product').select('*').ilike('category', `women's clothing`);
        if(data){
            setWomensProduct(data);
        }
        if(error){
            console.log(error);
            
        }
    }

    
    return {
        products, 
        getDataFromSupabase,
        filterData,
        getFilteredData,
        singleProduct,
        getSingleProduct,
        mensProduct,
        getMensClothing,
        womensProduct,
        getWomensClothing
    };
}