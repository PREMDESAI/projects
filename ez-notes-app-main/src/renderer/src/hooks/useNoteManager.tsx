// src/hooks/useNoteManager.ts
import { useState, useRef, useEffect } from 'react'
import { v4 as uuidv4 } from 'uuid'
import debounce from 'lodash.debounce'
import { Note } from '../global'
import { doc, setDoc } from 'firebase/firestore'
import { db } from '../firebase'

export function useNoteManager() {
  const [title, setTitle] = useState('')
  const [content, setContent] = useState('')
  const [currentNoteId, setCurrentNoteId] = useState<string | null>(null)
  const [notes, setNotes] = useState<Note[]>([])
  // const [notesSummary, setNotesSummary] = useState<Note[]>([])

  const titleRef = useRef(title)
  const contentRef = useRef(content)
  const currentNoteIdRef = useRef(currentNoteId)

  useEffect(() => {
    titleRef.current = title
  }, [title])

  useEffect(() => {
    contentRef.current = content
  }, [content])

  useEffect(() => {
    currentNoteIdRef.current = currentNoteId
  }, [currentNoteId])

  useEffect(() => {
    // Load existing notes on mount
    const loadNotes = async () => {
      try {
        const loadedNotes = await window.electronAPI.readAllNotes()
        setNotes(loadedNotes)
      } catch (error) {
        console.error('Error loading notes:', error)
      }
    }

    loadNotes()
  }, [])

  const handleSaveNote = async () => {
    const note = {
      id: currentNoteIdRef.current || uuidv4(),
      title: titleRef.current,
      content: contentRef.current
    }
    saveNoteContent(note.id, note.title, note.content)

    if (currentNoteIdRef.current) {
      // Update existing note
      await window.electronAPI.updateNote(note)
    } else {
      // Create new note
      await window.electronAPI.createNote(note)
      setCurrentNoteId(note.id)
      currentNoteIdRef.current = note.id
      setNotes((prevNotes) => [
        ...prevNotes,
        { id: note.id, title: note.title, content: note.content }
      ])
    }
  }

  const saveNoteContent = async (noteId, title, content) => {
    try {
      await setDoc(doc(db, 'notes', noteId), {
        title,
        content,
        lastEdited: new Date() // Optional
      })
    } catch (e) {
      console.error('Error writing document: ', e)
    }
  }

  const debouncedSave = useRef(
    debounce(() => {
      console.log('--FIREBASE CALL--')

      handleSaveNote()
    }, 1000)
  ).current

  useEffect(() => {
    return () => {
      debouncedSave.cancel()
    }
  }, [debouncedSave])

  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(e.target.value)
    debouncedSave()
  }

  const handleContentChange = (newContent: string) => {
    setContent(newContent)
    debouncedSave()
  }

  const handleSelectNote = async (id: string) => {
    try {
      const note = await window.electronAPI.readNote(id)
      if (note) {
        setTitle(note.title)
        setContent(note.content)
        setCurrentNoteId(note.id)
        currentNoteIdRef.current = note.id
      }
    } catch (error) {
      console.error('Error loading note:', error)
    }
  }

  return {
    title,
    content,
    notes,
    handleTitleChange,
    handleContentChange,
    handleSelectNote
  }
}
