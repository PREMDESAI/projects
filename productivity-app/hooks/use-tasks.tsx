// hooks/useTasks.ts
import { useState, useEffect } from 'react';
import { SelectTask } from '../../types'; // Adjust the import path as necessary

interface UseTasksReturn {
  tasks: SelectTask[];
  loading: boolean;
  error: string | null;
  refetch: () => void;
}

export const useTasks = (params: string): UseTasksReturn => {
  const [tasks, setTasks] = useState<SelectTask[]>([]);
  const [loading, setLoading] = useState<boolean>(true);
  const [error, setError] = useState<string | null>(null);

  const fetchTasks = async () => {
    const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL;
    setLoading(true);
    setError(null);
    let url = `${API_BASE_URL}/tasks`;
    if (params) {
      url += `?params=${params.toString()}`;
    }
    try {
      const res = await fetch(`${url}`, {
        cache: 'no-store',
      });
      if (!res.ok) {
        throw new Error('Failed to fetch tasks.');
      }
      const data = await res.json();
      console.log(data.tasks);

      setTasks(data.tasks);
    } catch (err: any) {
      console.error('Error fetching tasks:', err);
      setError(err.message || 'An error occurred.');
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    fetchTasks();
  }, []);

  const refetch = () => {
    fetchTasks();
  };

  return { tasks, loading, error, refetch };
};
