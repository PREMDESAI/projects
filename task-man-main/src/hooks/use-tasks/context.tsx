import React from 'react';

import { Task, TaskContextValues, TaskStatus } from './types';
import { TaskRepository } from '@/models/task';
import { STATUSES } from '@/constants';

export const TasksContext = React.createContext<TaskContextValues>({
  isLoading: false,
  tasksData: [],
  tasksView: [],
  statuses: [],

  addTask: () => Promise.resolve(),
  updateTask: () => Promise.resolve(),
  deleteTask: () => Promise.resolve(),
  filterTasks: () => null,
});

export const TasksProvider = ({ children }: React.PropsWithChildren) => {
  const [tasksData, setTasksData] = React.useState<Task[]>([]);
  const [tasksView, setTasksView] = React.useState<Task[]>([]);
  const [isLoading, setIsLoading] = React.useState(false);

  React.useEffect(() => {
    setIsLoading(true);
    TaskRepository.read()
      .then(setTasksData)
      .finally(() => setIsLoading(false));
  }, []);

  React.useEffect(() => {
    setTasksView([...tasksData]);
  }, [tasksData]);

  const addTask = React.useCallback(
    async (task: Omit<Task, 'id'>) => {
      setIsLoading(true);
      setTasksData([
        ...tasksData,
        {
          ...task,
          id: `TM-${tasksData.length + 1}`,
        },
      ]);

      const updatedTasks = await TaskRepository.create(task);
      setTasksData(updatedTasks);
      setIsLoading(false);
    },
    [tasksData]
  );

  const updateTask = React.useCallback(
    async (id: string, updatedTask: Omit<Task, 'id'>) => {
      setIsLoading(true);
      const indexToUpdate = tasksData.findIndex((task) => task.id === id);

      const clonedTasks = [...tasksData];
      clonedTasks.splice(indexToUpdate, 1);

      setTasksData([
        ...clonedTasks.slice(0, indexToUpdate),
        {
          id,
          ...updatedTask,
        },
        ...clonedTasks.slice(indexToUpdate),
      ]);

      const updatedTasks = await TaskRepository.update({ id, ...updatedTask });
      setTasksData(updatedTasks);
      setIsLoading(false);
    },
    [tasksData]
  );

  const deleteTask = React.useCallback(
    async (id: Task['id']) => {
      setIsLoading(true);

      const clonedTasks = [...tasksData];
      const indexToUpdate = tasksData.findIndex((task) => task.id === id);
      clonedTasks.splice(indexToUpdate, 1);
      setTasksData(clonedTasks);

      const updatedTasks = await TaskRepository.remove(id);
      setTasksData(updatedTasks);
      setIsLoading(false);
    },
    [tasksData]
  );

  const filterTasks = React.useCallback(
    ({ taskStatus, text }: { text?: string; taskStatus?: TaskStatus }) => {
      if (!taskStatus && !text) {
        setTasksView([...tasksData]);
        return;
      }
      const sanitizedText = text?.trim().toLowerCase();
      setTasksView([
        ...tasksData.filter(({ id, title, status }) => {
          let textMatch: boolean = false;
          let statusMatch: boolean = false;

          if (sanitizedText) {
            textMatch =
              id.toLowerCase().includes(sanitizedText) ||
              title.toLowerCase().includes(sanitizedText);
          }

          if (taskStatus) {
            statusMatch = taskStatus === status;
          }

          if (taskStatus && sanitizedText) return textMatch && statusMatch;

          if (taskStatus) return statusMatch;

          if (sanitizedText) return textMatch;
        }),
      ]);
    },
    [tasksData]
  );

  console.log(tasksData, tasksView);

  const value = React.useMemo<TaskContextValues>(
    () => ({
      tasksData,
      tasksView,
      statuses: STATUSES,
      isLoading,

      addTask,
      updateTask,
      deleteTask,
      filterTasks,
    }),
    [
      tasksData,
      tasksView,
      addTask,
      updateTask,
      deleteTask,
      filterTasks,
      isLoading,
    ]
  );

  return (
    <TasksContext.Provider value={value}>{children}</TasksContext.Provider>
  );
};
