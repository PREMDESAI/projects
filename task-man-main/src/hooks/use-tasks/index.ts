export { TasksProvider } from './context';
export { useTasks } from './useTasks';
export * from './types';
