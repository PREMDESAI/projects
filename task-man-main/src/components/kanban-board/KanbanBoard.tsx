import { Flex } from '@radix-ui/themes';

import { useTasks } from '@/hooks/use-tasks';
import { KanbanBoardPanels } from './KanbanBoardPanels';
import { styled } from '@stitches/react';

const StyledContainer = styled(Flex, {
  marginBottom: '32px',
  width: '100%',
  overflowX: 'auto',
  transition: '1s',
  variants: {
    isLoading: {
      true: {
        opacity: '0.5',
        pointerEvents: 'none',
      },
    },
  },
});

export const KanbanBoard: React.FC<React.PropsWithChildren> & {
  Panels: typeof KanbanBoardPanels;
} = ({ children }) => {
  const { isLoading } = useTasks();

  return (
    <StyledContainer direction='row' gap='5' isLoading={isLoading}>
      {children}
    </StyledContainer>
  );
};

KanbanBoard.Panels = KanbanBoardPanels;
