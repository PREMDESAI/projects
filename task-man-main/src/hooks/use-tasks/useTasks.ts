import React from 'react';

import { TasksContext } from './context';
import { TaskStatus } from './types';
import { debounce } from 'throttle-debounce';

export const useTasks = (status?: TaskStatus) => {
  const context = React.useContext(TasksContext);
  if (!context) {
    throw new Error('useTasks should be wrapped inside TasksProvider');
  }

  const { tasksView, filterTasks } = context;

  return {
    ...context,
    tasksView: status
      ? tasksView.filter((task) => task.status === status)
      : tasksView,
    filterTasks: debounce(500, filterTasks),
  };
};
