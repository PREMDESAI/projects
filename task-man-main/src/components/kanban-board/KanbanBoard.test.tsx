import { render } from '@testing-library/react';
import { describe, it } from 'vitest';

import { KanbanBoard } from './KanbanBoard';
import { TasksProvider } from '@/hooks/use-tasks';
import { MemoryRouter } from 'react-router-dom';

describe('KanbanBoard', () => {
  it('renders', async () => {
    const { container } = render(
      <MemoryRouter>
        <TasksProvider>
          <KanbanBoard>
            <KanbanBoard.Panels />
          </KanbanBoard>
        </TasksProvider>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
