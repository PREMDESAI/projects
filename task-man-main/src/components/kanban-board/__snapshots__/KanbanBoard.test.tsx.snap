// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KanbanBoard > renders 1`] = `
<div>
  <div
    class="rt-Flex rt-r-fd-row rt-r-gap-5 c-bFzNrF c-bFzNrF-bzJBWd-isLoading-true"
  >
    <div
      class="rt-Flex rt-r-fd-column rt-r-gap-3"
      style="height: calc(100vh - 128px - 32px); overflow-y: auto; background-color: rgb(245, 245, 250); border-radius: 4px; padding: 8px; flex-grow: 1; position: relative; max-width: 320px; min-width: 320px;"
    >
      <div
        class="rt-Flex rt-r-jc-space-between"
      >
        <p
          class="rt-Text"
          role="status"
          style="color: rgb(170, 170, 170); font-weight: bold;"
        >
          TO DO
           
        </p>
        <button
          class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-soft rt-IconButton"
          data-accent-color=""
          style="cursor: pointer;"
        >
          <svg
            fill="none"
            height="18"
            viewBox="0 0 15 15"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 2.75C8 2.47386 7.77614 2.25 7.5 2.25C7.22386 2.25 7 2.47386 7 2.75V7H2.75C2.47386 7 2.25 7.22386 2.25 7.5C2.25 7.77614 2.47386 8 2.75 8H7V12.25C7 12.5261 7.22386 12.75 7.5 12.75C7.77614 12.75 8 12.5261 8 12.25V8H12.25C12.5261 8 12.75 7.77614 12.75 7.5C12.75 7.22386 12.5261 7 12.25 7H8V2.75Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column rt-r-gap-3"
      style="height: calc(100vh - 128px - 32px); overflow-y: auto; background-color: rgb(245, 245, 250); border-radius: 4px; padding: 8px; flex-grow: 1; position: relative; max-width: 320px; min-width: 320px;"
    >
      <div
        class="rt-Flex rt-r-jc-space-between"
      >
        <p
          class="rt-Text"
          role="status"
          style="color: rgb(170, 170, 170); font-weight: bold;"
        >
          IN PROGRESS
           
        </p>
        <button
          class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-soft rt-IconButton"
          data-accent-color=""
          style="cursor: pointer;"
        >
          <svg
            fill="none"
            height="18"
            viewBox="0 0 15 15"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 2.75C8 2.47386 7.77614 2.25 7.5 2.25C7.22386 2.25 7 2.47386 7 2.75V7H2.75C2.47386 7 2.25 7.22386 2.25 7.5C2.25 7.77614 2.47386 8 2.75 8H7V12.25C7 12.5261 7.22386 12.75 7.5 12.75C7.77614 12.75 8 12.5261 8 12.25V8H12.25C12.5261 8 12.75 7.77614 12.75 7.5C12.75 7.22386 12.5261 7 12.25 7H8V2.75Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column rt-r-gap-3"
      style="height: calc(100vh - 128px - 32px); overflow-y: auto; background-color: rgb(245, 245, 250); border-radius: 4px; padding: 8px; flex-grow: 1; position: relative; max-width: 320px; min-width: 320px;"
    >
      <div
        class="rt-Flex rt-r-jc-space-between"
      >
        <p
          class="rt-Text"
          role="status"
          style="color: rgb(170, 170, 170); font-weight: bold;"
        >
          IN REVIEW
           
        </p>
        <button
          class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-soft rt-IconButton"
          data-accent-color=""
          style="cursor: pointer;"
        >
          <svg
            fill="none"
            height="18"
            viewBox="0 0 15 15"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 2.75C8 2.47386 7.77614 2.25 7.5 2.25C7.22386 2.25 7 2.47386 7 2.75V7H2.75C2.47386 7 2.25 7.22386 2.25 7.5C2.25 7.77614 2.47386 8 2.75 8H7V12.25C7 12.5261 7.22386 12.75 7.5 12.75C7.77614 12.75 8 12.5261 8 12.25V8H12.25C12.5261 8 12.75 7.77614 12.75 7.5C12.75 7.22386 12.5261 7 12.25 7H8V2.75Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="rt-Flex rt-r-fd-column rt-r-gap-3"
      style="height: calc(100vh - 128px - 32px); overflow-y: auto; background-color: rgb(245, 245, 250); border-radius: 4px; padding: 8px; flex-grow: 1; position: relative; max-width: 320px; min-width: 320px;"
    >
      <div
        class="rt-Flex rt-r-jc-space-between"
      >
        <p
          class="rt-Text"
          role="status"
          style="color: rgb(170, 170, 170); font-weight: bold;"
        >
          DONE
           
        </p>
        <button
          class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-soft rt-IconButton"
          data-accent-color=""
          style="cursor: pointer;"
        >
          <svg
            fill="none"
            height="18"
            viewBox="0 0 15 15"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 2.75C8 2.47386 7.77614 2.25 7.5 2.25C7.22386 2.25 7 2.47386 7 2.75V7H2.75C2.47386 7 2.25 7.22386 2.25 7.5C2.25 7.77614 2.47386 8 2.75 8H7V12.25C7 12.5261 7.22386 12.75 7.5 12.75C7.77614 12.75 8 12.5261 8 12.25V8H12.25C12.5261 8 12.75 7.77614 12.75 7.5C12.75 7.22386 12.5261 7 12.25 7H8V2.75Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
